packages/ckeditor5-theme-lark/theme/ckeditor5-link/linkform.css

```diff
@@ -4,58 +4,95 @@
  */
 
 @import "@ckeditor/ckeditor5-ui/theme/mixins/_dir.css";
+@import "@ckeditor/ckeditor5-ui/theme/mixins/_rwd.css";
 
-/*
- * Style link form differently when manual decorators are available.
- * See: https://github.com/ckeditor/ckeditor5-link/issues/186.
- */
-.ck.ck-link-form_layout-vertical {
-	padding: 0;
-	min-width: var(--ck-input-width);
+:root {
+	--ck-link-panel-width: 340px;
+	--ck-link-list-view-max-height: 240px;
+	--ck-link-list-view-icon-size: calc( var(--ck-icon-size) * 0.8); /* 0.8 = 16/20 cause default the icon size is 20px */
+}
 
-	& .ck-labeled-field-view {
-		margin: var(--ck-spacing-large) var(--ck-spacing-large) var(--ck-spacing-small);
+@mixin ck-media-phone {
+	:root {
+		--ck-link-panel-width: 300px;
+	}
+}
 
-		& .ck-input-text {
-			min-width: 0;
-			width: 100%;
-		}
+/* Generic class that wraps each form view. */
+.ck.ck-link__panel {
+	width: var(--ck-link-panel-width);
+
+	&:focus {
+		outline: none;
 	}
 
-	& > .ck-button {
-		padding: var(--ck-spacing-standard);
-		margin: 0;
-		width: 50%;
-		border-radius: 0;
+	/* Header */
+	& .ck-form__header {
+		padding: var(--ck-spacing-small) var(--ck-spacing-medium);
 
-		&:not(:focus) {
-			border-top: 1px solid var(--ck-color-base-border);
+		& > .ck:not(:first-child) {
+			margin-inline-start: var(--ck-spacing-small);
 		}
+	}
 
-		@mixin ck-dir ltr {
-			margin-left: 0;
+	/* Form */
+	& .ck-responsive-form.ck-link__form {
+		width: 100%;
+
+		& > * {
+			margin: 0;
+
+			@mixin ck-media-phone {
+				margin: var(--ck-spacing-large) var(--ck-spacing-medium) var(--ck-spacing-medium) var(--ck-spacing-medium);
+			}
 		}
 
-		@mixin ck-dir rtl {
-			margin-left: 0;
+		& > .ck:not(:first-child) {
+			margin-top: var(--ck-spacing-large);
+		}
 
-			&:last-of-type {
-				border-right: 1px solid var(--ck-color-base-border);
+		& .ck-labeled-field-view {
+			& input {
+				width: 100%;
 			}
 		}
-	}
 
-	/* Using additional `.ck` class for stronger CSS specificity than `.ck.ck-link-form > :not(:first-child)`. */
-	& .ck.ck-list {
-		margin: 0 var(--ck-spacing-large);
+		/* "Link" input field and the "Insert" button */
+		& .ck-link-and-submit {
+			& .ck-button-insert {
+				padding: var(--ck-spacing-tiny) var(--ck-spacing-large);
+				height: 30px;
```

Comment:
This should be somehow related to `--ck-font-size-base` so it scales with the whole UI.